Remove unnecessary content from the chef overview page

tas50 opened the pull request:
We should not be pointing brand new users to definitions and providers. There's no reason to even mention them in the getting started overview page. This removes a large amount of information, which rightfully makes Chef look simpler.

chef-delivery commented:
This PR has passed 'Verify' and is ready for review and approval!
  Use: '@delivery approve' when code review is complete.

TrevorBramble commented:
I suspect Google is a few iterations away from implementing self-aware artificial intelligence in their analytics snippet...

@delivery approve

chef-delivery commented:
Change: 3fa506fc-7ab4-4145-a6e4-7419b3c89752 approved by: @TrevorBramble

cwebberOps commented on _templates/layout.html:
```diff
@@ -45,33 +45,17 @@
-        document.location = url;
-      }
-    });
-  };
```
So lines 64-70 need to actually stay.

tas50 committed:
Remove definitions from the getting started overview

We should not be introducing anyone to definitions, certainly not in one
of the first documents they read.

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Remove references to providers from the overview

We talk about resources and there's no need to talk about legacy LWRPs
on this overview page.

Signed-off-by: Tim Smith <tsmith@chef.io>

davidwrede reviewed (approved):
👍  on removing the clutter

davidwrede commented:
![tenor-79408588](https://cloud.githubusercontent.com/assets/1634752/26168441/ee16e21a-3aee-11e7-8abe-5d0ce899e2ba.gif)
